handlers/oauth/oauth.go

```diff
@@ -0,0 +1,142 @@
+package oauth
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"math/rand"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgerrcode"
+	"github.com/markbates/goth"
+	"github.com/markbates/goth/gothic"
+	"github.com/markbates/goth/providers/twitter"
+	"github.com/speedrun-website/leaderboard-backend/model"
+)
+
+var twitterProvider = twitter.NewAuthenticate(
+	os.Getenv("TWITTER_OAUTH_KEY"),
+	os.Getenv("TWITTER_OAUTH_SECRET"),
+	os.Getenv("TWITTER_OAUTH_CALLBACK_URL"),
+)
+
+var completeUserAuth = gothic.CompleteUserAuth
+
+func InitializeProviders() {
+	enabledProviders := strings.Split(os.Getenv("ENABLED_PROVIDERS"), ",")
+	for _, provider := range enabledProviders {
+		cleanedProvider := strings.ToUpper(strings.TrimSpace(provider))
+		switch cleanedProvider {
+		case twitterProvider.Name():
```

Comment:
This is fine, but something that may clean it up:
If we have a `map[string]*Provider` where the `string` keys are values we expect to get from the env var, then we can both avoid the string transformation we have to do here, and avoid a potentially growing `switch` statement.